exonum/src/storage/proof_map_index/key.rs

```diff
@@ -152,9 +152,9 @@ pub enum ChildKind {
 }
 
 impl ::std::ops::Not for ChildKind {
-    type Output = ChildKind;
+    type Output = Self;
```

Comment:
Not related to your changes, but lets add `use std::ops` and change line above to the `impl ops::Not for ChildKind`.